lib/orcid/client.rb

```diff
@@ -8,13 +8,26 @@ class Client
     # Fetch the works for a researcher.
     # Model for the response: https://pub.orcid.org/v3.0/#!/Development_Public_API_v3.0/viewWorksv3
     def fetch_works(orcidid)
-      response = conn.get("/v3.0/#{orcidid[-19, 19]}/works")
+      get("/v3.0/#{base_orcidid(orcidid)}/works")
+    end
+
+    # Fetch the details for a work
+    def fetch_work(orcidid, put_code)
+      get("/v3.0/#{base_orcidid(orcidid)}/work/#{put_code}")
+    end
+
+    private
+
+    def get(url)
+      response = conn.get(url)
       raise "ORCID.org API returned #{response.status}" if response.status != 200
 
       JSON.parse(response.body).with_indifferent_access
     end
 
-    private
+    def base_orcidid(orcidid)
```

Comment:
we are going to need this kind of method when adding the ORCID-based query to the WoS client (was just taking a peek at that now) ... is there a better place to put it to make it easier to use in a different class?

Reply:
In general, an ORCID ID is considered the entire URL: `https://orcid.org/0000-0003-3437-349X`, not just `0000-0003-3437-349X`. That being said, I have no idea what WOS accepts.

Reply:
WoS wants the base identifier as well without the prefix

Reply:
Good catch. Extracting.